api/v1alpha1/servicesplitter_types_test.go

```diff
@@ -0,0 +1,317 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	capi "github.com/hashicorp/consul/api"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Test MatchesConsul for cases that should return true.
+func TestServiceSplitter_MatchesConsulTrue(t *testing.T) {
+	cases := map[string]struct {
+		Ours   ServiceSplitter
+		Theirs *capi.ServiceSplitterConfigEntry
+	}{
+		"empty fields": {
+			Ours: ServiceSplitter{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "name",
+				},
+				Spec: ServiceSplitterSpec{},
+			},
+			Theirs: &capi.ServiceSplitterConfigEntry{
+				Kind:        capi.ServiceSplitter,
+				Name:        "name",
+				Namespace:   "namespace",
+				CreateIndex: 1,
+				ModifyIndex: 2,
```

Comment:
nice!

Reply:
thank you! I pride myself in being good at copy-pasting code 😆